pepper/apps/trade_execution.c

```diff
@@ -44,16 +42,18 @@ void compute(struct In *input, struct Out *output) {
         struct Volume volume = volumes[index];
         
         if (fieldToInt(volume.sellVolume) > 0) {
-            // Verify volume has same ratio as prices
-            assert_zero((volume.buyVolume * prices[fieldToInt(order.buyToken)]) - (volume.sellVolume * prices[fieldToInt(order.sellToken)]));
+            // Verify volume has roughly same ratio as prices
+            field254 priceVolumeDelta = (volume.buyVolume * prices[fieldToInt(order.buyToken)]) - (volume.sellVolume * prices[fieldToInt(order.sellToken)]);
+            // Make sure |delta| < epsilon
+            assert_zero(isNegative(EPSILON - priceVolumeDelta) || isNegative(EPSILON + priceVolumeDelta));
```

Comment:
I guess we don't have the liberty of taking absolute values here? I don't believe the second condition here is correct. In the event that priceVolumeDelta is negative anything beyond 100, this would be true. Should also be an and condition in between.

```
|delta| < epsilon
```
is equivalent to 
```
- epsilon < delta < epsilon
```
which means we want
```
delta - epsilon < 0 and -delta - epsilon < 0
```
or, equivalently
```
delta - epsilon < 0 and delta + epsilon > 0
```

Reply:
Oh dang. Sorry, this assert zero reverses everything and this is fine.

Reply:
Hm, I think it's correct - I'd start from:
`-eps <= delta <= +eps`

This means
`delta - eps <= 0 and delta + eps >= 0`

or, equivalently,
`eps - delta >= 0 and eps + delta >= 0`

The assertion in Felix' code fails if either of the two is not satisfied (`isNegative`).